ddtrace/tracer/option.go

```diff
@@ -107,6 +113,10 @@ func defaults(c *config) {
 	}
 	if v := os.Getenv("DD_SERVICE"); v != "" {
 		c.serviceName = v
+		c.envService = v
```

Comment:
I saw your other comment but I don't get it. Why do we need this? It should be enough to just do:
```
if t.config.version != "" && span.Service != t.config.service {
```
The version applies to the core service. Regardless of where it is set (see other comment).

Reply:
`c.serviceName` can be set by options. We need to know when `span.Service == os.Getenv("DD_SERVICE")`